src/super_gradients/training/models/model_factory.py

```diff
@@ -135,6 +136,9 @@ def instantiate_model(
                 net.replace_head(new_num_classes=num_classes_new_head)
                 arch_params.num_classes = num_classes_new_head
 
+            class_names, image_processor = get_pretrained_processing_params(model_name, pretrained_weights)
+            net.set_dataset_processing_params(class_names, image_processor)
```

Comment:
Not a very big fan of this, because it is hard coding the "processing_params" as being `class_names` and `image_processor`.
But at the same time this might be all we will ever need, I feel like creating an abstraction for this would be overkill.
What do you think ?